transform/macros/repo_event_by_period.sql

```diff
@@ -0,0 +1,18 @@
+{% macro fact_event_type(event_type, period) -%} 
+SELECT
+    repo_id,
+    repo_name,
+    DATE_TRUNC('{{ period }}', event_date) AS event_period,
+    COUNT(*) AS event_count,
+    SUM(event_count) OVER (PARTITION BY repo_id ORDER BY event_period ASC) AS cumul_count
+FROM
+    {{ ref('stg_gharchive') }}
+WHERE
+    LOWER(event_type) = LOWER('{{ event_type }}')
```

Comment:
Ideally this kind of lowercase transformation is better handled in the staging layer, to allow for shared coding data conventions. Think we can move this logic there?